service/javadsl/integration-tests/src/test/scala/com/lightbend/lagom/it/JavadslErrorHandlingSpec.scala

```diff
@@ -53,69 +53,75 @@ import com.lightbend.lagom.javadsl.server.LagomServiceRouter
-        val result = client.streamRequest()
-          .invoke(Source.single(new MockRequestEntity("b", 2)).concat(Source.maybe).asJava)
-        try {
-          result.toCompletableFuture.get(10, TimeUnit.SECONDS)
-          throw sys.error("No error was thrown")
-        } catch {
-          case e: ExecutionException => e.getCause
-          case NonFatal(other)       => other
+      "handling errors with streamed request calls" should {
+        tests(new NamedCallId("streamRequest")) { implicit app => client =>
+          val result = client.streamRequest()
+            .invoke(Source.single(new MockRequestEntity("b", 2)).concat(Source.maybe).asJava)
+          try {
+            result.toCompletableFuture.get(10, TimeUnit.SECONDS)
+            throw sys.error("No error was thrown")
+          } catch {
+            case e: ExecutionException => e.getCause
+            case NonFatal(other)       => other
+          }
         }
       }
-    }
 
-    "handling errors with bidirectional streamed calls" should {
-      tests(new NamedCallId("bidiStream")) { implicit app => client =>
-        val result = client.bidiStream()
-          .invoke(Source.single(new MockRequestEntity("b", 2)).concat(Source.maybe).asJava)
-        try {
-          val resultSource = result.toCompletableFuture.get(10, TimeUnit.SECONDS)
-          Await.result(resultSource.asScala.runWith(Sink.ignore), 10.seconds)
-          throw sys.error("No error was thrown")
-        } catch {
-          case e: ExecutionException => e.getCause
-          case NonFatal(other)       => other
+      "handling errors with bidirectional streamed calls" should {
+        tests(new NamedCallId("bidiStream")) { implicit app => client =>
+          val result = client.bidiStream()
+            .invoke(Source.single(new MockRequestEntity("b", 2)).concat(Source.maybe).asJava)
+          try {
+            val resultSource = result.toCompletableFuture.get(10, TimeUnit.SECONDS)
+            Await.result(resultSource.asScala.runWith(Sink.ignore), 10.seconds)
+            throw sys.error("No error was thrown")
+          } catch {
+            case e: ExecutionException => e.getCause
+            case NonFatal(other)       => other
+          }
         }
       }
     }
   }
 
-  def tests(callId: CallId)(makeCall: Application => MockService => Throwable) = {
+  def tests(callId: CallId)(makeCall: Application => MockService => Throwable)(implicit httpBackend: HttpBackend) = {
     "handle errors in request serialization" in withClient(changeClient = change(callId)(failingRequestSerializer)) { implicit app => client =>
       makeCall(app)(client) match {
         case e: SerializationException =>
           e.errorCode should ===(TransportErrorCode.InternalServerError)
           e.exceptionMessage.detail should ===("failed serialize")
+        case ioe: java.io.IOException =>
+          ioe.printStackTrace()
+          throw ioe
```

Comment:
These can't just be bubbled up?

Reply:
Had to add all these to capture and trace the `IOException` when tracking down the issue found in https://github.com/lagom/lagom/pull/1054#issuecomment-339320868.

We could remove them but I'd rather keep these `case` in case (no pun intended) there's a regression.